capstone-ui/src/components/GroupHeader/GroupHeader.jsx

```diff
@@ -1,14 +1,54 @@
 import React, { useState } from "react";
 import "./GroupHeader.css";
 import axios from "axios";
+import { Link } from "react-router-dom";
+
+export default function GroupHeader({
+  username,
+  token,
+  profile,
+  group,
+  setShouldUpdateInbox,
+}) {
+  const [isMember, setIsMember] = useState(false);
+
+  const joinGroup = async (e) => {
+    let groupName =
+      e.target.parentNode.parentNode.childNodes[0].childNodes[0].childNodes[0]
+        .innerText;
+
+    e.preventDefault();
+    axios
+      .post("http://localhost:3001/user/joingroup", {
+        username: username,
+        groupName: groupName,
+      })
+      .then(function (response) {
+        alert(`You have joined ${groupName}!`);
+        setIsMember(true);
+      })
+      .catch(function (error) {
+        alert(`Error! ${error.message}`);
+      });
+    if (setShouldUpdateInbox && typeof setShouldUpdateInbox == "function") {
```

Comment:
setShouldUpdateInbox not be a function? Is this check necessary?

Reply:
I have this check to see if it is defined because sometimes I don't pass into the GroupHeader component a props called "setShouldUpdateInbox". There are only certain times that I want to update the parent component. Should I be going about this another way?